grow/pods/documents.py

```diff
@@ -264,6 +264,13 @@ def locales(self):
             if codes is None:
                 return []
             return locales.Locale.parse_codes(codes)
+        # If `$localization` is missing, use locales inferred from document
```

Comment:
One other thought, after re-reading my longer explanation above.

Instead of appending, we could just do a check. If the document doesn't specify `$locales` but the blueprint does, and if `$locale: de` is set on a document part (for example), we override the DE document. If the user sets `$locale: FOO` and `FOO` is not set in the blueprint or in the base part, then we raise an error. What do you think about that instead? That's probably the clearest and most consistent behavior.